pkg/cloudprovider/providers/vsphere/vclib/datacenter_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vclib
+
+import (
+	"context"
+	"testing"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/simulator"
+)
+
+func TestDatacenter(t *testing.T) {
+	ctx := context.Background()
+
+	// vCenter model + initial set of objects (cluster, hosts, VMs, network, datastore, etc)
+	model := simulator.VPX()
+
+	defer model.Remove()
+	err := model.Create()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := model.Service.NewServer()
+	defer s.Close()
+
+	avm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+
+	c, err := govmomi.NewClient(ctx, s.URL, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	vc := &VSphereConnection{GoVmomiClient: c}
+
+	_, err = GetDatacenter(ctx, vc, "enoent")
+	if err == nil {
```

Comment:
Can you please why ```enoent``` should return err as nil? Is this handled differently in the simulator? Since I cannot find it here https://github.com/vmware/govmomi/blob/master/vcsim/README.md/#usage

Reply:
It is the opposite, these tests are making sure that `enoent` returns an error that is **not** nil.  Look at the line below, calls `t.Error` if `err == nil`.  The name `enoent` does not exist anywhere in the inventory, the tests are expecting these calls to fail, providing coverage of the error paths in `datacenter.go`.

Reply:
Sounds good to me